### PR TITLE
Augment MLB endpoint for SportsDataIO

### PR DESCRIPTION
**Pull request by cds95**
## Description

Add additional capabilities to the MLB SportsDataIO endpoint.

## Changes

Augments the MLB endpoint for SportsDataIO with the following

 - Allow filtering game schedule by `teamID`
 - Add `onlyShowGameIDs` flag to `schedule` endpoint to only have `gameIDs` in the response.
 - Set `AwayTeamMoneyLine`, `HomeTeamMoneyLine`, `AwayTeamRuns` and `HomeTeamRuns` to `-1` if a the game's status is not `Final`.

## Steps to Test

1. Run adapter locally and test using the below request payloads and verify responses.

Filter by team
```
{
  "id": 1,
  "data": {
    "date": "2017-JUL-31",
    "sport": "mlb",
    "endpoint": "schedule",
    "teamID": "BAL"
    }
}
```

Only show game IDs

```
{
  "id": 1,
  "data": {
    "date": "2017-JUL-31",
    "sport": "mlb",
    "endpoint": "schedule",
    "onlyShowGameIDs": true
  }
}
```

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/sources/sportsdataio/src/sport/mlb/utils.ts [hunk 1]
```diff
@@ -33,3 +33,11 @@ export const getGamesByDate = async (date: string, config: Config): Promise<Game
...
   )
 }
+
+export const getGamesByTeam = (games: GameResponse[], teamID: string): GameResponse[] => {
+  return games.filter(
+    (game) =>
+      game.AwayTeam.toLowerCase() === teamID.toLowerCase() ||
+      game.HomeTeam.toLowerCase() === teamID.toLowerCase(),
+  )
+}
```
Comment (austinborn) [inline packages/sources/sportsdataio/src/sport/mlb/utils.ts:43]:
```suggestion
export const getGamesByTeam = (games: GameResponse[], teamID: string): GameResponse[] => {
  const lowercaseTeamID =  teamID.toLowerCase()
  return games.filter(
    (game) =>
      game.AwayTeam.toLowerCase() === lowercaseTeamID ||
      game.HomeTeam.toLowerCase() === lowercaseTeamID,
  )
}
```

**Review by austinborn — COMMENTED** [review]
Changeset?

**Review by austinborn — APPROVED** [review]

**Review by austinborn — APPROVED** [review]

File: packages/sources/sportsdataio/README.md [hunk 2]
```diff
@@ -129,9 +130,45 @@ Get the MLB schedule on a given date
...
-| Required? | Name |                Description                 | Options | Defaults to |
-| :-------: | :--: | :----------------------------------------: | :-----: | :---------: |
-|    ✅     | date | The date games to query were/are played on |         |             |
+| Required? |      Name      |                                      Description                                      |          Options           | Defaults to |
+| :-------: | :------------: | :-----------------------------------------------------------------------------------: | :------------------------: | :---------: |
+|    ✅     |      date      |                      The date games to query were/are played on                       |                            |             |
+|           |     teamID     |                           The team ID to filter by e.g LAD                            | See list of team IDs below |             |
+|           | onlyShowGameID | Set to true if response should only contain game IDs instead of the whole game object |                            |             |
+
+#### List of team IDs
```
Comment (boxhock) [inline packages/sources/sportsdataio/README.md:139]: Who will maintain this list? Can we link to any docs from SDIO instead of taking this responsibility?

**Comment by eddiejhong** [comment]
Closing this PR for now as this work has been deprioritized.

**Pull request closed without merging**